### PR TITLE
Refactor code generation

### PR DESCRIPTION
**Pull request by azazeal**
This PR refactors the codebase in order to address our internal build issue. It additionally adds checks to the `Makefile` that ensure that code is generated via suitable versions of `protoc-gen-go` and `protoc-gen-go-grpc`.

### CODE REVIEW

**Comment by maraino** [comment]
A change like this will cause backward incompatibility problems between open-source step-ca and majordomo.

File: majordomo_grpc.pb.go [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Code generated by protoc-gen-go-grpc. DO NOT EDIT.
 // versions:
-// - protoc-gen-go-grpc v1.3.0
+// - protoc-gen-go-grpc v1.2.0
```
Comment (maraino) [inline majordomo_grpc.pb.go:3]: Can you upgrade your `protoc-gen-go-grpc` to at least v1.3.0

Comment (azazeal) [reply majordomo_grpc.pb.go]: I've added Makefile checks for this purpose.

File: Makefile [hunk 2]
```diff
@@ -2,6 +2,16 @@
 Q=$(if $V,,@)
 SRC=$(shell find . -type f -name '*.go')
 
+# protoc-gen-go constraints
+GEN_GO_BIN ?= protoc-gen-go
+GEN_GO_MIN_VERSION ?= 1.31.0
+GEN_GO_VERSION ?= $(shell $(GEN_GO_BIN) --version | awk --field-separator=' v' '{print $$NF}')
```
Comment (maraino) [inline Makefile:8]: This didn't work for me on macOS nor in Ubuntu 23.04, but this works:

```suggestion
GEN_GO_VERSION ?= $(shell $(GEN_GO_BIN) --version | awk -F ' v' '{print $$NF}')
```

File: Makefile [hunk 3]
```diff
@@ -2,6 +2,16 @@
...
+
+# protoc-gen-go-grpc constraints
+GEN_GRPC_BIN ?= protoc-gen-go-grpc
+GEN_GRPC_MIN_VERSION ?= 1.3.0
+GEN_GRPC_VERSION ?= $(shell $(GEN_GRPC_BIN) --version | awk --field-separator=' ' '{print $$NF}')
```
Comment (maraino) [inline Makefile:13]:
```suggestion
GEN_GRPC_VERSION ?= $(shell $(GEN_GRPC_BIN) --version | awk -F ' ' '{print $$NF}')
```

**Review by maraino — CHANGES_REQUESTED** [review]
Seems to work fine, but I've added a couple of suggestions to be able to run `make generate`.

File: Makefile (hunk 2 above)
Comment (azazeal) [reply Makefile:8]: weird, I'm on ubuntu myself and the action's also running on ubuntu in CI.

File: Makefile (hunk 3 above)
Comment (azazeal) [reply Makefile:13]: weird, I'm on ubuntu myself and the action's also running on ubuntu in CI.

**Review by maraino — APPROVED** [review]

**Pull request merged**